geniza/corpus/tests/test_corpus_templates.py

```diff
@@ -78,10 +79,53 @@ def test_viewer_annotations(self, client, document, unpublished_editions):
         response = client.get(document.get_absolute_url())
         assertContains(
             response,
-            '<input type="checkbox" class="toggle" id="transcription-on" checked="true" aria-label="show transcription">',
+            '<input type="checkbox" class="toggle" id="transcription-on" data-ittpanel-target="toggle" data-action="ittpanel#clickToggle" checked="true" aria-label="show transcription">',
             html=True,
         )
 
+        # add a footnote with a digital translation
+        hebrew = SourceLanguage.objects.get(name="Hebrew")
+        unpublished_editions.languages.add(hebrew)
+        translation = Footnote.objects.create(
+            content_object=document,
+            source=unpublished_editions,
+            doc_relation=[Footnote.DIGITAL_TRANSLATION],
+        )
+        response = client.get(document.get_absolute_url())
+        # should contain header label for this translation
+        assertContains(
+            response,
+            f'<span data-transcription-target="translationShortLabel">Translation: {unpublished_editions.all_authors()} {unpublished_editions.all_languages()}</span>',
+            html=True,
+        )
+
+        # add an annotation to the translation so we can show the next part of the template
+        # (looping over canvases to display translation content)
+        Annotation.objects.create(
+            footnote=translation,
+            content={
+                "body": [
+                    {
+                        "value": "test",
+                    },
+                ],
+                "target": {
+                    "source": {
+                        "id": "fake_canvas",
+                    },
+                },
+                "motivation": ["sc:supplementing", "translating"],
+            },
+        )
+        response = client.get(document.get_absolute_url())
+        print(response.content)
+        # should contain rtl and language code, since this is a translation to hebrew
+        # NOTE: had to use BeautifulSoup here due to inexplicable failures with assertContains
```

Comment:
Yes, that was primarily what I was using, and I couldn't get it to work (after a long time!). My only remaining thought is that it might have had to do with some Unicode in the `formatted_display`, but didn't think it was worth investing any more time.

Reply:
ugh. ok, understood, makes sense. (maybe worth noting it might be due to unicode)